src/js/components/productSupplier/ProductSupplierListTable.jsx

```diff
@@ -1,27 +1,103 @@
-import React, { useMemo } from 'react';
+import React, { useMemo, useState } from 'react';
 
 import PropTypes from 'prop-types';
+import { RiDeleteBinLine, RiPencilLine } from 'react-icons/ri';
+import { useSelector } from 'react-redux';
 
+import productSupplierApi from 'api/services/ProductSupplierApi';
 import DataTable, { TableCell } from 'components/DataTable';
 import DateCell from 'components/DataTable/DateCell';
 import PreferenceTypeColumn from 'components/productSupplier/PreferenceTypeColumn';
+import { PRODUCT_SUPPLIER_URL } from 'consts/applicationUrls';
 import useProductSupplierListTableData from 'hooks/list-pages/productSupplier/useProductSupplierListTableData';
+import ActionDots from 'utils/ActionDots';
+import CustomModal from 'utils/CustomModal';
+import { hasPermissionsToProductSourceActions } from 'utils/permissionUtils';
 import StatusIndicator from 'utils/StatusIndicator';
 import Translate from 'utils/Translate';
 import ListTableTitleWrapper from 'wrappers/ListTableTitleWrapper';
 import ListTableWrapper from 'wrappers/ListTableWrapper';
 
 const ProductSupplierListTable = ({ filterParams }) => {
+  const [isDeleteConfirmationOpened, setIsDeleteConfirmationOpened] = useState(false);
+  const [selectedProductSupplierId, setSelectedProductSupplierId] = useState(null);
+
+  const { currentUser } = useSelector((state) => ({
+    currentUser: state.session.user,
+  }));
+
+  const closeDeleteConfirmationModal = () => {
+    setIsDeleteConfirmationOpened(false);
+  };
+
+  const deleteProductSupplier = async () => {
+    try {
+      await productSupplierApi.deleteProductSupplier(selectedProductSupplierId);
+    } finally {
+      setIsDeleteConfirmationOpened(false);
+    }
+  };
+
+  const deleteConfirmationModalButtons = [
+    {
+      variant: 'transparent',
+      defaultLabel: 'Cancel',
+      label: 'Cancel',
+      onClick: closeDeleteConfirmationModal,
+    },
+    {
+      variant: 'danger',
+      defaultLabel: 'Delete',
+      label: '',
+      onClick: deleteProductSupplier,
+    },
+  ];
+
+  const getActions = (id) => [
+    {
+      defaultLabel: 'Edit',
+      label: 'react.productSupplier.edit.label',
+      leftIcon: <RiPencilLine />,
+      onClick: () => {
+        window.location = PRODUCT_SUPPLIER_URL.edit(id);
+      },
+    },
+    {
+      defaultLabel: 'Delete Product Source',
+      label: 'react.productSupplier.delete.label',
+      leftIcon: <RiDeleteBinLine />,
+      variant: 'danger',
+      onClick: () => {
+        setIsDeleteConfirmationOpened(true);
+        setSelectedProductSupplierId(id);
+      },
+    },
+  ];
+
```

Comment:
here, the actions are not that big yet, but just wanted to mention that we used to add the `useMemo` for actions.
The `actions` used to be a const, not a method, and the eventual id has only been passed to the `onClick` by ActionDots component and its `id` prop., e.g.:
```js
Cell: row => (
        <ActionDots
          dropdownPlacement="right"
          dropdownClasses="action-dropdown-offset"
          actions={findActions(actions, row, { customFilter: customActionFilter, highestRole })}
          id={row.original.id}
        />),
```
and an action:
```js
{
    defaultLabel: 'Publish stock list',
    label: 'react.stocklists.publish.label',
    leftIcon: <RiFile3Line />,
    isPublished: false,
    onClick: publishStocklists,
    minimumRequiredRole: 'Admin',
},
```

In this case as the actions are not that big, I don't know if re-rendering of them will impact the performance much (I mean, if adding the `useMemo` would help or rather make the performance get worse).

Reply:
I think it's better to leave to the current version because now it's just a ternary operator for now. In `findActions` we have much more complexity. Additionally, we can't pass `row` to the `hasPermissionsToProductSourceActions` and this function needs to be refactored especially for this one use case